fragment_internal_test.go

```diff
@@ -713,6 +713,58 @@ func BenchmarkFragment_ImportValue(b *testing.B) {
 	}
 }
 
+// BenchmarkFragment_RepeatedSmallImports tests the situation where updates are
+// constantly coming in across a large column space so that each fragment gets
+// only a few updates during each import.
+//
+// We test a variety of combinations of the number of separate updates(imports),
+// the number of bits in the import, the number of rows in the fragment (which
+// is a pretty good proxy for fragment size on disk), and the MaxOpN on the
+// fragment which controls how many set bits occur before a snapshot is done. If
+// the number of bits in a given import is greater than MaxOpN, bulkImport will
+// always go through the standard snapshotting import path.
+func BenchmarkFragment_RepeatedSmallImports(b *testing.B) {
+	for _, numUpdates := range []int{100} {
+		for _, bitsPerUpdate := range []int{100, 1000} {
+			for _, numRows := range []int{1000, 100000, 1000000} {
+				// build the update data set all at once - this will get applied
```

Comment:
hmm. suggestion: possibly move the creation of the slices out (we could just do a 100*1000 slice for each outside the loop, and reslice them at the beginning of the loop), and b.StopTimer()/b.StartTimer() around the population of the bits, just so they aren't being counted. Although I guess that doesn't matter as much since there's a b.Run() below?

Reply:
nevermind just noticed this is changed by next commit.